.github/workflows/build.yaml

```diff
@@ -0,0 +1,204 @@
+        if: always()
+        run: sha256sum ola-*.tar.gz
+      - name: Upload source tree artifact
+        uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: ola-${{ matrix.id }}-source-tree
+          path: ola-${{ matrix.id }}-source-tree.tar.gz
+      - name: Upload built artifact
+        if: matrix.task == 'distcheck' || matrix.task == 'dist'
+        uses: actions/upload-artifact@v3
+        with:
+          name: ola-${{ matrix.id }}-dist
+          path: |
+            ola-*.tar.gz
+            !ola-${{ matrix.id }}-source-tree.tar.gz
+  coverage:
+    name: 'Coverage for ${{ matrix.id }}'
+    needs: build
+    if: "always()" # Run if some builds fail but ensure they all complete first
+    container: debian:stable
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { id: "coverage-debian-stable-amd64-gcc" }
+    steps:
+      - name: Download built source tree archive
+        uses: actions/download-artifact@v3
+        with:
+          name: ola-${{ matrix.id }}-source-tree
+          path: .
+      - name: SHA256 artifact archive
+        run: sha256sum ola-${{ matrix.id }}-source-tree.tar.gz
+      - name: Unarchive artifacts and delete archive
+        shell: bash
+        run: |
+          tar -xvzf ola-${{ matrix.id }}-source-tree.tar.gz .
+          rm ola-${{ matrix.id }}-source-tree.tar.gz
+      - name: Display structure of extracted files
+        if: env.ACTIONS_STEP_DEBUG == 'true'
+        run: ls -alR
+      - name: Update package database
+        run: apt-get update -y
+      - name: Install coverage tools
+        run: apt-get -y install curl gcc python3-pip
+      - name: Install Python coverage tools
+        run: python3 -m pip install --no-input gcovr
+      - name: Install build dependencies
+        shell: bash
+        run: |
+          apt-get -y install libcppunit-dev uuid-dev libncurses5-dev \
+            libmicrohttpd-dev protobuf-compiler python3-protobuf \
+            libprotobuf-dev libprotoc-dev zlib1g-dev libftdi-dev \
+            libusb-1.0-0-dev liblo-dev libavahi-client-dev python3-numpy \
+            default-jdk-headless maven
+      - name: Generate coverage
+        shell: bash
+        run: make coverage VERBOSE=1
+      - name: Upload coverage to Coveralls
+        uses: coverallsapp/github-action@v2
+        with:
+          # Coveralls GitHub action does not support its own format
+          # see: https://github.com/coverallsapp/github-action/issues/104
+          # file: coverage.coveralls.json
+          file: coverage/coverage.cobertura.xml
+          format: cobertura
+          flag-name: ${{ matrix.id }}
+      - name: Upload coverage artifacts
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: ola-${{ matrix.id }}-coverage
+          path: coverage/
+  verify-trees:
+    name: 'Verify trees for ${{ matrix.id }}'
```

Comment:
I'd rather this didn't run while its returning what I suspect are false positives. Can we either comment it out or removed the commits to a future PR (but don't bother with the latter unless it's trivial).

I think the issue is either this build being more complete, or us not properly cleaning up the contents of nodist, but I'm not sure which is technically the problem.

Reply:
Removing the commits will not be trivial; I'll see if I can mark as OK-to-fail or I'll comment it out. We did a similar thing for spellintian (still runs but issues only warnings and returns success).

Reply:
Fixed.

Reply:
Build succeeds now!

Reply:
I quite liked the spellintian solution, or leaving the matcher but matching as info, but this is also fine.